### ISSUE TITLE
Support running `aleph-client` on macOS

### ISSUE THREAD

**Issue by hoh** [aleph-im/aleph-client#40]
Many developers use macOS, and it would be convenient for this project to run natively on macOS.

This may require installing dependencies using [homebrew](https://brew.sh/).

**Comment by hoh** [comment]
Can we add automated tests on GitHub actions for this ?
https://github.blog/changelog/2021-08-16-github-actions-macos-11-big-sur-is-generally-available-on-github-hosted-runners/

**Comment by MHHukiewitz** [comment]
There's also the full list of hostable runners: https://github.com/actions/virtual-environments

**Comment by hoh** [comment]
Related to https://github.com/aleph-im/aleph-client/pull/79

**Comment by hoh** [comment]
@mrhouzlane what remains to be done for macOS support ? 

I see the GitBook documentation still needs updating: https://aleph-im.gitbook.io/aleph-client/quickstart/getting-setup

What else ?

**Comment by hoh** [comment]
While wheels are now also built for macOS thankts to #83 , tests are not yet running on macOS via GitHub Actions.

**Comment by MHHukiewitz** [comment]
> While wheels are now also built for macOS thankts to https://github.com/aleph-im/aleph-client/pull/83 , tests are not yet running on macOS via GitHub Actions.
Did they ever run in the first place? ^^'

**Comment by hoh** [comment]
Yes, tests run in a Docker image via this workflow: https://github.com/aleph-im/aleph-client/blob/master/.github/workflows/test-pytest.yml

**Comment by mrhouzlane** [comment]
but macos doesn't allow macos containers
macos in a container doesn't exist as Apple really doesn't want their os virtualized or containerized, so I don't think the test with Docker image for macos would be possible

**Comment by hoh** [comment]
You can run tests on the Github Actions macOS runner in a way similar to [building the wheels](https://github.com/aleph-im/aleph-client/blob/master/.github/workflows/test-build.yml), without a Docker container.
